src/autogluon/cloud/utils/ray_aws_iam.py

```diff
@@ -47,20 +51,48 @@
         {
             "Effect": "Allow",
             "Action": ["ec2:CreateKeyPair"],
-            "Resource": f"arn:aws:ec2:*:{POLICY_ACCOUNT_PLACE_HOLDER}:key-pair/*",
+            "Resource": [f"arn:aws:ec2:*:{POLICY_ACCOUNT_PLACE_HOLDER}:key-pair/*"],
         },
         {
             "Effect": "Allow",
             "Action": [
                 "iam:GetInstanceProfile",
                 "iam:CreateInstanceProfile",
                 "iam:CreateRole",
+                "iam:CreatePolicy",
+                "iam:DeletePolicy",
                 "iam:GetRole",
                 "iam:AttachRolePolicy",
+                "iam:DetachRolePolicy",
                 "iam:AddRoleToInstanceProfile",
                 "iam:PassRole",
             ],
             "Resource": ["*"],
```

Comment:
The `CreatePolicy`/`DeletePolicy` roles are too powerful to give permissions like these: is it possible to narrow-down `Resource` to something more specific than `*`?

Reply:
Good point. I'll restrict these two to follow a specific pattern